extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/theta/SketchBufferAggregator.java

```diff
@@ -113,4 +110,28 @@ public void close()
     unions.clear();
   }
 
+  @Override
+  public void relocate(int oldPosition, int newPosition, ByteBuffer newBuffer)
```

Comment:
I think this method should be be`relocate(int[] newPositions, ByteBuffer newBuffer)` and implementation should discard all old unions and build new ones for each new position and store in the unions map